app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localBackend.ts

```diff
@@ -168,21 +181,37 @@ export class LocalBackend implements Partial<backend.Backend> {
                 state: {
                     type: backend.ProjectState.opened,
                 },
-            })
+            }
         }
     }
 
     /** Prepare a project for execution.
      *
      * @throws An error if the JSON-RPC call fails. */
-    async openProject(projectId: backend.ProjectId): Promise<void> {
+    async openProject(
+        projectId: backend.ProjectId,
+        _body: backend.OpenProjectRequestBody | null,
+        title: string | null
+    ): Promise<void> {
         LocalBackend.currentlyOpeningProjectId = projectId
-        const project = await this.projectManager.openProject({
-            projectId,
-            missingComponentAction: projectManager.MissingComponentAction.install,
-        })
-        if (LocalBackend.currentlyOpeningProjectId === projectId) {
-            LocalBackend.currentlyOpenProject = { id: projectId, project }
+        try {
+            const project = await this.projectManager.openProject({
+                projectId,
+                missingComponentAction: projectManager.MissingComponentAction.install,
+            })
+            if (LocalBackend.currentlyOpeningProjectId === projectId) {
+                LocalBackend.currentlyOpenProject = { id: projectId, project }
+            }
+            return
+            // This is UNSAFE. It is assumed that all errors thrown are instances of `Error`
+            // and therefore have a string property called `message`.
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (e: any) {
```

Comment:
The same comment as above - if its unsafe, we should either remove it, or write safety guarantees.